sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java

```diff
@@ -1318,7 +1318,7 @@ public void flush() throws IOException {
       if (getFooter() != null) {
         writer.println(getFooter());
       }
-      writer.close();
+      writer.flush();
```

Comment:
Ah didn't realize that close() would swallow exceptions. Could you add a comment here so that this does not get reverted. Also, please check if there are other places in the JDK if we do that same.